src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -513,6 +491,42 @@ contract GameToken is ERC721BaseToken, WithMinter, IGameToken {
         return newId;
     }
 
+    /// @dev Check if a withdrawal is allowed.
+    /// @param from The address requesting the withdrawal.
+    /// @param gameId The id of the GAME token to withdraw assets from.
+    function _check_withdrawal_authorized(address from, uint256 gameId) internal view {
+        require(from != address(0), "SENDER_ZERO_ADDRESS");
+        require(from == _withdrawalOwnerOf(gameId), "LAST_OWNER_NOT_EQUAL_SENDER");
+    }
+
+    /// @dev Get the address allowed to withdraw assets from the GAME token.
+    /// If too many assets in GAME, block.gaslimit won't allow detroy and withdraw in 1 tx.
+    /// A game owner may destroy their GAME token, then withdraw assets in a later tx (even
+    /// though ownerOf(id) would be address(0) after burning.)
+    /// @param id The id of the GAME token to query.
+    /// @return the address of the owner before burning.
+    function _withdrawalOwnerOf(uint256 id) internal view returns (address) {
+        uint256 data = _owners[_storageId(id)];
+        if ((data & BURNED_FLAG) == BURNED_FLAG) {
+            return address(data);
+        }
+        return address(0);
+    }
+
+    /// @dev A GameToken-specific implementation which handles versioned tokenIds.
+    /// @param id The tokenId to get the owner of.
+    /// @return The address of the owner.
+    function _ownerOf(uint256 id) internal view override returns (address) {
```

Comment:
What about ERC721BaseToken _ownerOf implementation.
we should maybe remove the storageId mechanism and instead move that into the sub contract
because havign _ownerOf use storageId introduce the risk of _ownerOf returning an owner for all tokenid that share the storage id.
We could fix that by documenting, but I think it might be better to have a basic _ownerOf that do not introduce the issue and then it is up to the sub contract, to override _ownerOf and take the responsibility like you do in GameToken

Reply:
I was thinking along similar lines but was more focused on the `_storageId()` function. As it is now, The only way to introduce that risk is for someone to either modify `_storageId()` (it currently just returns the id passed to it) or override it(as we've done in the GameToken).

But I think your suggestion is cleaner. Remove the storageId concept from the Base token contract completely.
easy enough to override _ownerOf if versioned tokenIds are desired, less risk overall, and as a bonus, we don't have a function `_storageId` in the base token that just returns the value passed to it